windows/src/engine/keyman32/K32_load.cpp

```diff
@@ -75,11 +75,65 @@ BOOL GetKeyboardFileName(LPSTR kbname, LPSTR buf, int nbuf)
   return n;
 }
 
+
+BOOL LoadlpKeyboardCore(int i)
+{
+  PKEYMAN64THREADDATA _td = ThreadGlobals();
+  if (!_td) return FALSE;
+  if (_td->lpKeyboards[i].coreKeyboard) return TRUE;
+  
+
+  if (_td->lpActiveKeyboard == &_td->lpKeyboards[i]) _td->lpActiveKeyboard = NULL;  // I822 TSF not working
+
+  char buf[256];
+  if (!GetKeyboardFileName(_td->lpKeyboards[i].Name, buf, 255)) return FALSE;
+  size_t charSize = strlen(buf) + 1;
+
+  LPWSTR keyboardPath = new WCHAR[charSize];
+
+  // Convert char* string to a wchar_t* string.
+  size_t convertedChars = 0;
+  mbstowcs_s(&convertedChars, keyboardPath, charSize, buf, _TRUNCATE);
```

Comment:
Not sure what you mean by considerations on codepage an you please expand?. With buffer size if the "buf" is long enough (which has been so far in the life of keyman) then making the WCHAR the size of the number of characters in buf will be sufficent. see https://docs.microsoft.com/en-us/cpp/text/how-to-convert-between-various-string-types?view=msvc-160

Reply:
Per https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/mbstowcs-mbstowcs-l?view=msvc-160
> mbstowcs uses the current locale for any locale-dependent behavior

Also, we have `strtowstr` defined in kmprocess.cpp which I think does what we want? (It may not be completely ideal as a function but better to use what the existing code is using I guess.

Reply:
I do accept that `strtowstr` does not consider locale either so that could be a future concern.